### ISSUE TITLE
Advice - Adding functionality to generated SDK's

### ISSUE THREAD

**Issue by jhitchcock** [swagger-api/swagger-codegen#3520]
I want to add functionality to the generated SDK's. Obviously I can add this into the templates and build my own swagger service, but I also wish to keep it up to date with at least the major swagger releases. 

This is custom business logic, for example, extra methods to get an API key from a web login etc. 

We support Java, JS, PHP

Does anyone have any advice on a nice way to do this? If not, how a feature to do this might look and and I can look into contributing it. 

I guess one main problem is that every language may do this differently, if some kind of dependency is used. 

Anyway, may be an interesting discussion! 

Thanks!

**Comment by wing328** [comment]
> extra methods to get an API key from a web login etc.

Can that web login be documented as another endpoint in OpenAPI/Swagger spec?

**Comment by jhitchcock** [comment]
Nah, its a bit more complicated, a bit of a 'dance' with a few systems. 

Here's another example; 

In order to add payment cards, one must obtain a token from a service, and open a web popup with a specific URL. 

In the generated libraries, we would like a method such as 'getPaymentCardUrl' that does this 'dance' and returns the URL to open.. so its pretty custom logic. I'd create a class in the templates called utils or something, and add the methods, Maybe this is already the best way!

**Comment by wing328** [comment]
> I'd create a class in the templates called utils or something, and add the methods, Maybe this is already the best way!

I would probably do the same as the logic is complicated and hard to be generalized into the spec.
